Client/game_sa/CFileLoaderSA.cpp

```diff
@@ -201,7 +206,7 @@ RpAtomic* CFileLoader_SetRelatedModelInfoCB(RpAtomic* atomic, SRelatedModelInfo*
     return atomic;
 }
 
-CEntitySAInterface* CFileLoader_LoadObjectInstance(const char* szLine)
+CBuildingSAInterface* CFileLoader_LoadObjectInstance(const char* szLine)
```

Comment:
I think current variant describes this function better. CFileLoader_LoadObjectInstance always returns a building object and not a another class object inherited from CEntitySAInterface

Reply:
LoadObjectInstance can return either CBuilding or CDummyObject as can be verified by looking at reversed SA. You can pass model 3797 into this function and it will return CDummyObject which is not a building.